src/test/java/com/upserve/uppend/performance/MultiKeyTest.java

```diff
@@ -0,0 +1,88 @@
+package com.upserve.uppend.performance;
+
+import com.upserve.uppend.*;
+import com.upserve.uppend.util.SafeDeleting;
+import org.junit.*;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.*;
+
+import static com.upserve.uppend.performance.StreamTimerMethods.*;
+
+public class MultiKeyTest {
+
+    private static final int numPartitions = 12;
+    private static final int hashSize = 32;
+    private static final int values = 1_000;
+    private static final int blobsPerBlock = 64;
+    private static final int keyCount = 1_000;
+
+    private final Path path = Paths.get("build/test/tmp/performance/multi_key");
+    private AppendOnlyStore appendOnlyStore;
+    private final String[] keys = new String[keyCount]; // Just use a single key - it will be hashed appropriately for partition and hashsize
+    private final int repeats = 5;
+
+    private final Supplier<LongStream> parallelStream = () -> Arrays.stream(keys).flatMapToLong(key -> appendOnlyStore.read(key, key).mapToLong(bytes -> bytes.length).parallel()).parallel();
+    private final Supplier<LongStream> sequentialStream = () -> Arrays.stream(keys).flatMapToLong(key -> appendOnlyStore.read(key, key).mapToLong(bytes -> bytes.length).sequential()).sequential();
+
+    @Before
+    public void loadStore() throws IOException {
+        SafeDeleting.removeTempPath(path);
+
+        appendOnlyStore = new AppendOnlyStoreBuilder()
+                .withPartitionSize(numPartitions)
```

Comment:
This method name confuses me. Is it the _size_ of the partition or the _number_ of partitions? Should the method be renamed `withPartitionCount`?

Reply:
Ugh - your right - lets hope the IDE does most of the work on this one...